qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/diagnostics/QuteDiagnosticsInExpressionTest.java

```diff
@@ -706,6 +706,100 @@ public void parameterSuperType() {
 						DiagnosticSeverity.Error));
 	}
 
+	@Test
+	public void switchExpressionTypeMismatchSwitch() {

```

Comment:
Move your tests in a new class `QuteDiagnosticsInExpressionWithSwitchWhenSectionTest`